### PR TITLE
Introducing RDMACM support in pyverbs

### PR DESCRIPTION
**Pull request by noaos**
The following patches add support for RDMACM in pyverbs. Currently
only synchronous data path is supported (creation using rdma_create_ep).
Testing infrastructure for RDMACM is also added as well as a synchronous
traffic test.

### CODE REVIEW

**Comment by jgunthorpe** [comment]
Lets rebase this please

**Comment by noaos** [comment]
Done.

File: tests/test_rdmacm.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+
+class CMTestCase(RDMATestCase):
+    def setUp(self):
+        if self.dev_name is not None:
+            net_name = self.get_net_name(self.dev_name)
+            try:
+                self.ip_addr = self.get_ip_address(net_name)
+            except KeyError:
+                raise unittest.SkipTest('Device {} has no net interface'
```
Comment (rleon) [inline tests/test_rdmacm.py]: has no net -> doesn't have net

File: tests/base.py [hunk 2]
```diff
@@ -144,6 +151,51 @@ def _add_gids_per_device(self, ctx, dev):
...
+    def __init__(self, src=ZERO_ADDR, dst=ZERO_ADDR, port=PORT):
+        """
+        :param src: Local address to bind to (for passive side)
+        :param dst: Destination address to connect (for active side)
+        :param port: Port number of the address
+        """
+        self.is_server = True if dst == ZERO_ADDR else False
+        self.qp_init_attr = None
+        self.msg_size = 1024
+        self.num_msgs = 100
```
[the remark this answers is not in the document]
Comment (maxim-chi) [reply tests/base.py]: [maximc@c-134-3-160-161 rdma-core-26.0]$ python tests/run_tests.py -v
test_rdmacm_sync_traffic (tests.test_rdmacm.CMTestCase) ... ok       

----------------------------------------------------------------------
Ran 1 test in 2.666s

OK
[maximc@c-134-3-160-161 rdma-core-26.0]$ python tests/run_tests.py -v
test_rdmacm_sync_traffic (tests.test_rdmacm.CMTestCase) ... ok

----------------------------------------------------------------------
Ran 1 test in 4.847s

OK
[maximc@c-134-3-160-161 rdma-core-26.0]$ python tests/run_tests.py -v
test_rdmacm_sync_traffic (tests.test_rdmacm.CMTestCase) ... ok

----------------------------------------------------------------------
Ran 1 test in 2.775s

OK
[maximc@c-134-3-160-161 rdma-core-26.0]$ python tests/run_tests.py -v
test_rdmacm_sync_traffic (tests.test_rdmacm.CMTestCase) ... ok

----------------------------------------------------------------------
Ran 1 test in 9.582s

OK
[maximc@c-134-3-160-161 rdma-core-26.0]$ python tests/run_tests.py -v
test_rdmacm_sync_traffic (tests.test_rdmacm.CMTestCase) ... ok

----------------------------------------------------------------------
Ran 1 test in 5.131s

OK
[maximc@c-134-3-160-161 rdma-core-26.0]$ python tests/run_tests.py -v
test_rdmacm_sync_traffic (tests.test_rdmacm.CMTestCase) ... ok

----------------------------------------------------------------------
Ran 1 test in 2.340s

OK

Tried on all devices supported, I'll reduce anyway

File: tests/base.py [hunk 3]
```diff
@@ -26,6 +29,10 @@
 RNR_RETRY = 7
 TIMEOUT = 14
 
+# for rdmacm
+PORT = '7471'
+ZERO_ADDR = '0.0.0.0'
```
[the remark this answers is not in the document]
Comment (maxim-chi) [reply tests/base.py]: defines above will be used in my next submission

File: tests/base.py [hunk 4]
```diff
@@ -144,6 +151,51 @@ def _add_gids_per_device(self, ctx, dev):
             self._add_gids_per_port(ctx, dev, port+1)
 
 
+class CMResources:
+    """
+    CMResources class is a base aggregator object which contains basic
+    resources for RDMA CM communication.
+    """
+    def __init__(self, src=ZERO_ADDR, dst=ZERO_ADDR, port=PORT):
```
[the remark this answers is not in the document]
Comment (maxim-chi) [reply tests/base.py]: node param used both as src and dst address 
AddrInfo obj wraps
int rdma_getaddrinfo(char *node, char *service, struct rdma_addrinfo *hints, struct rdma_addrinfo **res)
C examples uses node param as src or dst address

**Comment by noaos** [comment]
PR was updated.

Thanks,
Noa

**Comment by rleon** [comment]
It worked on my devices,

[leonro@server build]$ ./bin/run_tests.py 
...........................................s.................
----------------------------------------------------------------------
Ran 61 tests in 52.201s

OK (skipped=1)

Thanks

**Pull request merged**